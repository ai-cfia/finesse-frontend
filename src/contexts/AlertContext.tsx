import React, {
  createContext,
  useContext,
  useState,
  type ReactNode,
} from "react";

// Type for the alert context state
interface AlertContextState {
  alertMessage: string | null;
  setAlertMessage: React.Dispatch<React.SetStateAction<string | null>>;
}

// Create the context with a default value
const AlertContext = createContext<AlertContextState>({
  alertMessage: null,
  setAlertMessage: () => {},
});

// Create a provider component for the alert context
export const AlertProvider: React.FC<{ children: ReactNode }> = ({
  children,
}) => {
  const [alertMessage, setAlertMessage] = useState<string | null>(null);

  return (
    <AlertContext.Provider value={{ alertMessage, setAlertMessage }}>
      {children}
    </AlertContext.Provider>
  );
};

// Custom hook to use the alert context
export const useAlert = (): AlertContextState => {
  const context = useContext(AlertContext);
  if (context === null) {
    throw new Error("useAlert must be used within an AlertProvider");
  }
  return context;
};
