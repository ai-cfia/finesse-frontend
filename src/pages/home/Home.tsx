import { useEffect } from "react";
import { PingBackend } from "../../api/useApiUtil";
import AlertBanner from "../../components/alert_banner/AlertBanner";
import DebugButton from "../../components/debug_button/DebugButton";
import { DebugPanel } from "../../components/debug_panel/DebugPanel";
import Header from "../../components/header/Header";
import CFIALogo from "../../components/logo/CFIALogo";
import { SearchBar } from "../../components/searchbar/SearchBar";
import { useAlert } from "../../contexts/AlertContext";
import { useData } from "../../contexts/DataContext";
import { environment } from "../../environments/environment";
import styles from "../home/Home.module.css";

const Home: React.FC = () => {
  const { currentSearchSource } = useData();
  const { setAlertMessage } = useAlert();

  useEffect(() => {
    console.log("currentSearchSource:", currentSearchSource);

    if (
      process.env.REACT_APP_BACKEND_URL === null ||
      process.env.REACT_APP_BACKEND_URL === ""
    ) {
      setAlertMessage(
        "Warning: Backend URL is not set, frontend is misconfigured.",
      );
    } else {
      PingBackend()
        .then((response) => {
          if (response === "ok") {
            console.log("Ping response:", response);
          } else {
            setAlertMessage(
              "Warning: Initializing ping request to backend failed.",
            );
          }
        })
        .catch(() => {
          setAlertMessage(
            "Warning: Initializing ping request to backend failed.",
          );
        });
    }
  }, [currentSearchSource, setAlertMessage]);

  return (
    <div className={styles.layout}>
      <Header />
      <AlertBanner />
      <DebugButton
        style={{
          cursor: "pointer",
          zIndex: 1001,
          position: "absolute",
          left: "10px",
          marginTop: "10px",
        }}
      />
      <DebugPanel />
      <div className="logo-container">
        <CFIALogo />
      </div>
      <div className="searchBar-container">
        <SearchBar />
      </div>
      <div
        style={{
          display: "flex",
          textAlign: "center",
          marginTop: 20,
          flexDirection: "column",
          alignItems: "center",
          color: "grey",
        }}
      >
        <span>Empowering agency&apos;s users with precision search.</span>
        <span style={{ marginTop: 10 }}>
          Équiper les utilisateurs de l&apos;agence avec la recherche de
          précision.
        </span>
      </div>
      <div className={styles.versionText}>
        {environment.version !== "" ? "v" + environment.version : ""}
      </div>
    </div>
  );
};

export default Home;
