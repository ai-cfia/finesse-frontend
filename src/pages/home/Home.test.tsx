import "@testing-library/jest-dom";
import {
  fireEvent,
  render,
  screen,
  type RenderResult,
} from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { AlertProvider } from "../../contexts/AlertContext";
import { DataProvider } from "../../contexts/DataContext";
import { LayoutProvider } from "../../contexts/LayoutContext";
import Home from "./Home"; // Adjust the import path as needed.

jest.mock("../../api/useApiUtil", () => ({
  PingBackend: async () => await Promise.resolve("ok"),
}));

describe("Home", () => {
  const renderHomePage = (): RenderResult => {
    return render(
      <BrowserRouter>
        <AlertProvider>
          <DataProvider>
            <LayoutProvider>
              <Home />
            </LayoutProvider>
          </DataProvider>
        </AlertProvider>
      </BrowserRouter>,
    );
  };

  it("renders the Home page correctly", () => {
    const { asFragment } = renderHomePage();
    expect(
      screen.getByText(/Empowering agency's users with precision search./),
    ).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  describe("Environment variable REACT_APP_BACKEND_URL", () => {
    it("shows a warning message when REACT_APP_BACKEND_URL is not set", () => {
      process.env.REACT_APP_BACKEND_URL = "";
      renderHomePage();
      expect(
        screen.getByText(
          /Warning: Backend URL is not set, frontend is misconfigured./,
        ),
      ).toBeInTheDocument();
    });

    it("does not show a warning message when REACT_APP_BACKEND_URL is set", () => {
      process.env.REACT_APP_BACKEND_URL = "http://example.com";
      renderHomePage();
      expect(
        screen.queryByText(
          /Warning: Backend URL is not set, frontend is misconfigured./,
        ),
      ).not.toBeInTheDocument();
    });
  });

  describe("Debug functionalities", () => {
    it("shows the debug button when REACT_APP_DEBUG_MODE is true", () => {
      process.env.REACT_APP_DEBUG_MODE = "true";
      const { asFragment } = renderHomePage();
      expect(screen.getByTestId("debug-button")).toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    });

    it("does not show the debug button when REACT_APP_DEBUG_MODE is false", () => {
      process.env.REACT_APP_DEBUG_MODE = "false";
      const { asFragment } = renderHomePage();
      expect(screen.queryByTestId("debug-button")).not.toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    });

    it("toggles the debug panel when the debug button is clicked", () => {
      process.env.REACT_APP_DEBUG_MODE = "true";
      const { asFragment } = renderHomePage();
      fireEvent.click(screen.getByTestId("debug-button"));
      expect(screen.getByTestId("debug-panel")).toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
      fireEvent.click(screen.getByTestId("debug-button"));
      expect(screen.queryByTestId("debug-panel")).not.toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
