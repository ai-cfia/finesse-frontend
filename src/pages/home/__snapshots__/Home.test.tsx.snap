// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Debug functionalities does not show the debug button when REACT_APP_DEBUG_MODE is false 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <header
      class="header"
      role="banner"
    >
      <div
        class="headerContainer"
      >
        <nav>
          <ul
            class="headerNavList"
          >
            <li
              class="headerNavLeftMargin"
            >
              <a
                href="https://inspection.canada.ca/"
                title="ACIA | CFIA"
              >
                <img
                  alt="CFIA logo"
                  aria-label="Link to CFIA | Lien à l'ACIA"
                  src="CFIA_Banner.png"
                  style="height: 50px;"
                />
              </a>
            </li>
            <li
              class="headerRightText"
            >
              Alpha Version
            </li>
          </ul>
        </nav>
      </div>
    </header>
    <div
      class="logo-container"
    >
      <div>
        <a
          data-testid="cfia-logo"
          href="/"
          title="ACIA | CFIA"
        >
          <img
            alt="CFIA logo"
            class="githubLogo"
            src="CFIA_SmartSearch_Logo.png"
          />
        </a>
      </div>
    </div>
    <div
      class="searchBar-container"
    >
      <form
        class="form-wrapper"
      >
        <div
          class="input-wrapper"
        >
          <svg
            data-testid="search-icon"
            fill="currentColor"
            height="1em"
            id="fa-arrow-right"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <input
            data-testid="search-input"
            placeholder="Type to search..."
            value=""
          />
          <button
            class="button"
            data-testid="submit-button"
            type="submit"
          >
            <svg
              fill="currentColor"
              height="1em"
              id="fa-arrow-right"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(5, 72, 108);"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
              />
            </svg>
          </button>
        </div>
      </form>
    </div>
    <div
      style="display: flex; text-align: center; margin-top: 20px; flex-direction: column; align-items: center; color: grey;"
    >
      <span>
        Empowering agency's users with precision search.
      </span>
      <span
        style="margin-top: 10px;"
      >
        Équiper les utilisateurs de l'agence avec la recherche de précision.
      </span>
    </div>
    <div
      class="versionText"
    >
      v0.2.4
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home Debug functionalities shows the debug button when REACT_APP_DEBUG_MODE is true 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <header
      class="header"
      role="banner"
    >
      <div
        class="headerContainer"
      >
        <nav>
          <ul
            class="headerNavList"
          >
            <li
              class="headerNavLeftMargin"
            >
              <a
                href="https://inspection.canada.ca/"
                title="ACIA | CFIA"
              >
                <img
                  alt="CFIA logo"
                  aria-label="Link to CFIA | Lien à l'ACIA"
                  src="CFIA_Banner.png"
                  style="height: 50px;"
                />
              </a>
            </li>
            <li
              class="headerRightText"
            >
              Alpha Version
            </li>
          </ul>
        </nav>
      </div>
      <svg
        data-testid="debug-button"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: black; cursor: pointer; z-index: 1001; position: absolute; margin: 10px;"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
        />
      </svg>
    </header>
    <div
      class="logo-container"
    >
      <div>
        <a
          data-testid="cfia-logo"
          href="/"
          title="ACIA | CFIA"
        >
          <img
            alt="CFIA logo"
            class="githubLogo"
            src="CFIA_SmartSearch_Logo.png"
          />
        </a>
      </div>
    </div>
    <div
      class="searchBar-container"
    >
      <form
        class="form-wrapper"
      >
        <div
          class="input-wrapper"
        >
          <svg
            data-testid="search-icon"
            fill="currentColor"
            height="1em"
            id="fa-arrow-right"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <input
            data-testid="search-input"
            placeholder="Type to search..."
            value=""
          />
          <button
            class="button"
            data-testid="submit-button"
            type="submit"
          >
            <svg
              fill="currentColor"
              height="1em"
              id="fa-arrow-right"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(5, 72, 108);"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
              />
            </svg>
          </button>
        </div>
      </form>
    </div>
    <div
      style="display: flex; text-align: center; margin-top: 20px; flex-direction: column; align-items: center; color: grey;"
    >
      <span>
        Empowering agency's users with precision search.
      </span>
      <span
        style="margin-top: 10px;"
      >
        Équiper les utilisateurs de l'agence avec la recherche de précision.
      </span>
    </div>
    <div
      class="versionText"
    >
      v0.2.4
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home Debug functionalities toggles the debug panel when the debug button is clicked 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <header
      class="header"
      role="banner"
    >
      <div
        class="headerContainer"
      >
        <nav>
          <ul
            class="headerNavList"
          >
            <li
              class="headerNavLeftMargin"
            >
              <a
                href="https://inspection.canada.ca/"
                title="ACIA | CFIA"
              >
                <img
                  alt="CFIA logo"
                  aria-label="Link to CFIA | Lien à l'ACIA"
                  src="CFIA_Banner.png"
                  style="height: 50px;"
                />
              </a>
            </li>
            <li
              class="headerRightText"
            >
              Alpha Version
            </li>
          </ul>
        </nav>
      </div>
      <svg
        data-testid="debug-button"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: black; cursor: pointer; z-index: 1001; position: absolute; margin: 10px;"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
        />
      </svg>
      <div
        class="debug-panel"
        data-testid="debug-panel"
      >
        <h4
          style="color: black;"
        >
          Debug Panel
        </h4>
        <div
          class="radio-container"
        >
          <label
            class="radio-option"
          >
            <input
              checked=""
              type="radio"
              value="ailab"
            />
            <span
              class="radio-label"
            >
              Use AI Lab search
            </span>
          </label>
          <label
            class="radio-option"
          >
            <input
              type="radio"
              value="azure"
            />
            <span
              class="radio-label"
            >
              Use Azure AI Search
            </span>
          </label>
          <label
            class="radio-option"
          >
            <input
              type="radio"
              value="simulated"
            />
            <span
              class="radio-label"
            >
              Use Simulated Data
            </span>
          </label>
        </div>
      </div>
    </header>
    <div
      class="logo-container"
    >
      <div>
        <a
          data-testid="cfia-logo"
          href="/"
          title="ACIA | CFIA"
        >
          <img
            alt="CFIA logo"
            class="githubLogo"
            src="CFIA_SmartSearch_Logo.png"
          />
        </a>
      </div>
    </div>
    <div
      class="searchBar-container"
    >
      <form
        class="form-wrapper"
      >
        <div
          class="input-wrapper"
        >
          <svg
            data-testid="search-icon"
            fill="currentColor"
            height="1em"
            id="fa-arrow-right"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <input
            data-testid="search-input"
            placeholder="Type to search..."
            value=""
          />
          <button
            class="button"
            data-testid="submit-button"
            type="submit"
          >
            <svg
              fill="currentColor"
              height="1em"
              id="fa-arrow-right"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(5, 72, 108);"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
              />
            </svg>
          </button>
        </div>
      </form>
    </div>
    <div
      style="display: flex; text-align: center; margin-top: 20px; flex-direction: column; align-items: center; color: grey;"
    >
      <span>
        Empowering agency's users with precision search.
      </span>
      <span
        style="margin-top: 10px;"
      >
        Équiper les utilisateurs de l'agence avec la recherche de précision.
      </span>
    </div>
    <div
      class="versionText"
    >
      v0.2.4
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home Debug functionalities toggles the debug panel when the debug button is clicked 2`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <header
      class="header"
      role="banner"
    >
      <div
        class="headerContainer"
      >
        <nav>
          <ul
            class="headerNavList"
          >
            <li
              class="headerNavLeftMargin"
            >
              <a
                href="https://inspection.canada.ca/"
                title="ACIA | CFIA"
              >
                <img
                  alt="CFIA logo"
                  aria-label="Link to CFIA | Lien à l'ACIA"
                  src="CFIA_Banner.png"
                  style="height: 50px;"
                />
              </a>
            </li>
            <li
              class="headerRightText"
            >
              Alpha Version
            </li>
          </ul>
        </nav>
      </div>
      <svg
        data-testid="debug-button"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: black; cursor: pointer; z-index: 1001; position: absolute; margin: 10px;"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
        />
      </svg>
    </header>
    <div
      class="logo-container"
    >
      <div>
        <a
          data-testid="cfia-logo"
          href="/"
          title="ACIA | CFIA"
        >
          <img
            alt="CFIA logo"
            class="githubLogo"
            src="CFIA_SmartSearch_Logo.png"
          />
        </a>
      </div>
    </div>
    <div
      class="searchBar-container"
    >
      <form
        class="form-wrapper"
      >
        <div
          class="input-wrapper"
        >
          <svg
            data-testid="search-icon"
            fill="currentColor"
            height="1em"
            id="fa-arrow-right"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <input
            data-testid="search-input"
            placeholder="Type to search..."
            value=""
          />
          <button
            class="button"
            data-testid="submit-button"
            type="submit"
          >
            <svg
              fill="currentColor"
              height="1em"
              id="fa-arrow-right"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(5, 72, 108);"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
              />
            </svg>
          </button>
        </div>
      </form>
    </div>
    <div
      style="display: flex; text-align: center; margin-top: 20px; flex-direction: column; align-items: center; color: grey;"
    >
      <span>
        Empowering agency's users with precision search.
      </span>
      <span
        style="margin-top: 10px;"
      >
        Équiper les utilisateurs de l'agence avec la recherche de précision.
      </span>
    </div>
    <div
      class="versionText"
    >
      v0.2.4
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home renders the Home page correctly 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <header
      class="header"
      role="banner"
    >
      <div
        class="headerContainer"
      >
        <nav>
          <ul
            class="headerNavList"
          >
            <li
              class="headerNavLeftMargin"
            >
              <a
                href="https://inspection.canada.ca/"
                title="ACIA | CFIA"
              >
                <img
                  alt="CFIA logo"
                  aria-label="Link to CFIA | Lien à l'ACIA"
                  src="CFIA_Banner.png"
                  style="height: 50px;"
                />
              </a>
            </li>
            <li
              class="headerRightText"
            >
              Alpha Version
            </li>
          </ul>
        </nav>
      </div>
      <svg
        data-testid="debug-button"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: black; cursor: pointer; z-index: 1001; position: absolute; margin: 10px;"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
        />
      </svg>
    </header>
    <div
      class="logo-container"
    >
      <div>
        <a
          data-testid="cfia-logo"
          href="/"
          title="ACIA | CFIA"
        >
          <img
            alt="CFIA logo"
            class="githubLogo"
            src="CFIA_SmartSearch_Logo.png"
          />
        </a>
      </div>
    </div>
    <div
      class="searchBar-container"
    >
      <form
        class="form-wrapper"
      >
        <div
          class="input-wrapper"
        >
          <svg
            data-testid="search-icon"
            fill="currentColor"
            height="1em"
            id="fa-arrow-right"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <input
            data-testid="search-input"
            placeholder="Type to search..."
            value=""
          />
          <button
            class="button"
            data-testid="submit-button"
            type="submit"
          >
            <svg
              fill="currentColor"
              height="1em"
              id="fa-arrow-right"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(5, 72, 108);"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
              />
            </svg>
          </button>
        </div>
      </form>
    </div>
    <div
      style="display: flex; text-align: center; margin-top: 20px; flex-direction: column; align-items: center; color: grey;"
    >
      <span>
        Empowering agency's users with precision search.
      </span>
      <span
        style="margin-top: 10px;"
      >
        Équiper les utilisateurs de l'agence avec la recherche de précision.
      </span>
    </div>
    <div
      class="versionText"
    >
      v0.2.4
    </div>
  </div>
</DocumentFragment>
`;
